src/Differ.php

```diff
@@ -19,6 +18,12 @@
  */
 final class Differ
 {
+    public const OLD                     = 0;
```

Comment:
is it part of API?
shouldn't this be marked as `@internal`  (or moved to dedicated, `@internal` enum-class ?)

Reply:
The reason why this component did not use constants before was the fact that, at least with PHP 5, the indirection of using a constant (every time the constant is used its value needs to be fetched) was significantly slower than to use the literal `0`, `1`, etc.

I am sure that this has gotten better with PHP 7, especially when OpCache is used (as it simply replaces the usage of a constant with the constant's value during bytecode generation). Using an enum class, however, is something that cannot be optimized away.

Reply:
Also, since we're now requiring PHP 7.1, those constants should be declared `private`.

Reply:
It is API. This package allows for 3rd party output builders which will rely on either the `magic` values or these `const`'s.
I see no need for `internal` markings as such, moving to dedicated `class` seems odd to me as form OOP view a class is not a collection of `const` pretending to be an `enum` for no other reason than it can be.

Reply:
Nevermind, due to output builders it is part of the API.

Reply:
great, thanks for explanation !